### ISSUE TITLE
pylint treats 'from compression import zstd' as third-party-import on python 3.14

### ISSUE THREAD

**Issue by plywakd** [pylint-dev/pylint#10867]
### Bug description

```python
from compression import zstd
from typing import Optional
from unittest.mock import Mock

import numpy as np


def main():
    print("Hello from python3-14!")


if __name__ == "__main__":
    main()

For above sample code pylint raises C0411 error and treats compression as third-party-import, but it is already available in standard package in python 3.14. Skip the unused imports, it's just a showcase of imports ordering.
```

### Configuration

```ini
Pyproject.toml:

[project]
name = "python3-14-pylint"
version = "0.1.0"
description = "Bug report"
readme = "README.md"
requires-python = ">=3.14"
dependencies = [
    "numpy>=2.4.2",
]

[dependency-groups]
dev = [
    "pylint==4.0.5",
]
```

### Command used

```shell
uv run pylint main.py
```

### Pylint output

```python
************* Module main
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:8:0: C0116: Missing function or method docstring (missing-function-docstring)
main.py:2:0: C0411: standard import "typing.Optional" should be placed before third party import "compression.zstd" (wrong-import-order)
main.py:3:0: C0411: standard import "unittest.mock.Mock" should be placed before third party import "compression.zstd" (wrong-import-order)
main.py:1:0: W0611: Unused zstd imported from compression (unused-import)
main.py:2:0: W0611: Unused Optional imported from typing (unused-import)
main.py:3:0: W0611: Unused Mock imported from unittest.mock (unused-import)
main.py:5:0: W0611: Unused numpy imported as np (unused-import)
```

### Expected behavior

No error C0411 raised for compression imports when using python 3.14

### Pylint version

```shell
4.0.5, also tried this on 4.0.4 with same result
```

### OS / Environment

cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. This is due to isort or the isort configuration. So it's going to be fixed in isort or by a change of isort configuration and there's nothing that should change in pylint's code to fix this.

**Comment by DanielNoord** [comment]
We released a new version of isort that should fix this!
